src/etc/inc/services.inc

```diff
@@ -1087,16 +1087,67 @@ function services_dhcpdv6_configure($blacklist = array(), $verbose = false)
             $ifcfgipv6 = Net_IPv6::getNetmask($ifcfgipv6, 64);
             $trackifname = $config['interfaces'][$ifname]['track6-interface'];
             $trackcfg = $config['interfaces'][$trackifname];
+            $ifcfgipv6arr = array();
             $pdlen = calculate_ipv6_delegation_length($trackifname);
             $ifcfgipv6arr =explode(":", $ifcfgipv6);
             $dhcpdv6cfg[$ifname] = array();
             $dhcpdv6cfg[$ifname]['enable'] = true;
             /* range */
-            $ifcfgipv6arr[7] = "1000";
-            $dhcpdv6cfg[$ifname]['range'] = array();
-            $dhcpdv6cfg[$ifname]['range']['from'] = Net_IPv6::compress(implode(":", $ifcfgipv6arr));
-            $ifcfgipv6arr[7] = "2000";
-            $dhcpdv6cfg[$ifname]['range']['to'] = Net_IPv6::compress(implode(":", $ifcfgipv6arr));
+            if(!isset($config['interfaces'][$ifname]['dhcpd6track6allowoverride']) || (isset($config['interfaces'][$ifname]['dhcpd6track6allowoverride']) && !isset($config['dhcpdv6'][$ifname]['enable']))) {
```

Comment:
this whole block could be clearer, let me try to merge tomorrow. nice work so far!

Reply:
I'm sure there's a better way, but I can't see it at present.